File: El Mamoun/Lv 1/Problem 32.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
/*
 Problem 32: Write a program to ask the user to enter:• Number • M
 Then Print the Number^M
 */


namespace Problem_32
{
    internal class Program
    {
        static int ReadPostiveNumber(string Message)
        {

            int Number = 0;
            do
            {
                Console.WriteLine(Message);
                Number = Convert.ToInt32(Console.ReadLine());

            } while (Number < 0);

            return Number;
        }
        static int Power_OfM(int Number, int Base)
        {
            int P = 1;


            if (Base == 0)
                return 1;

            for (int Counter = 1; Counter <= Base; Counter++)
            {
                P = Number * P;
            }
            return P;

        }


        static void Main(string[] args)
        {


            Console.WriteLine(Power_OfM(
            ReadPostiveNumber("Please Enter a Number?: "), ReadPostiveNumber("Please Enter a Base?: ")));
            Console.ReadKey();


        }
    }
}

File: El Mamoun/Lv 1/Problem 32.cpp
#include<iostream>
using namespace std;

/*
 Problem 32: Write a program to ask the user to enter:• Number • M
 Then Print the Number^M
 */

int ReadNumber()
{
	int number = 0;

	cout << "Please Enter a Number? ";
	cin >> number;

	return number;
}

int ReadPower()
{
	int number = 0;

	cout << "Please Enter a Power? ";
	cin >> number;

	return number;
}

int PowerOfM(int Number, int M)
{
	if (M == 0)
	{
		return 1;
	}
		
	int P = 1;
	for (int i = 1; i <= M; i++)
	{
		P = P * Number;
	}

	return P;
}

int main()
{
	cout << "Result = " << PowerOfM(ReadNumber(), ReadPower()) << endl;

	return 0;
}